src/lib.rs

```diff
@@ -103,7 +111,13 @@
        html_root_url = "https://docs.rs/uuid")]
 
 #![deny(warnings)]
-#![no_std]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+// serde links to std, so go ahead an pull in our own std
+// support in those situations as well.
+#[cfg(any(feature = "std",
+          feature = "serde"))]
+extern crate core;
```

Comment:
I don't think will work when compiling with `no_std`. Instead we might need to do something like:

```rust
#[cfg(feature = "std")]
extern crate std as core;
```

Reply:
Actually, let's try and sort out our `std` features. Right now, if you specify `--no-default-features --features serde` you'll get a compile error from duplicate uses of `core`.

We never need anything from `serde` that uses `std` though, so we could specify it in our `Cargo.toml` as:

```toml
serde = { version = "1.0.16", optional = true, default-features = false }
```

And then include core in `lib.rs` as:

```rust
#[cfg(feature = "std")]
extern crate std as core;
```

Then `no_std` users can use `uuid` with `serde`.